### PR TITLE
feat(unlock-app): Events social cues

### PR DESCRIPTION
**Pull request by julien51**
# Description

We're adding social details about the hosts (Farcaster or Lens profile). This is the first step on a much longer set of features.

Some examples:

Screenshot 2024-04-02 at 6 09 33 PM

Screenshot 2024-04-02 at 6 09 52 PM

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [X] I have performed a self-review of my own code
- [X] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/hooks/useEventOrganizers.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            locksByNetwork[lock.network!] = []
+          }
+          locksByNetwork[lock.network!].push(lockAddress)
+        }
+      })
+      const eventOrganizers: string[] = []
+      await Promise.all(
+        Object.keys(locksByNetwork).map(async (network: string) => {
+          const locks = locksByNetwork[network]
+          const locksWithManagers = await service.locks(
```
Comment (clemsos) [inline unlock-app/src/hooks/useEventOrganizers.ts:43]: can you get just the `managers` field here ?

File: unlock-app/src/hooks/useEventOrganizers.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            {
+              networks: [network],
+            }
+          )
+          locksWithManagers.forEach((lock) => {
+            lock.lockManagers.forEach((manager: string) => {
+              if (eventOrganizers.indexOf(manager) == -1) {
+                eventOrganizers.push(manager)
+              }
+            })
```
Comment (clemsos) [inline unlock-app/src/hooks/useEventOrganizers.ts:59]: maybe use `reduce` ?

File: unlock-app/src/hooks/useSocials.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      profileMetadata
+      profileName
+      profileTokenAddress
+      profileTokenId
+      profileTokenIdHex
+      profileTokenUri
+      profileUrl
+      twitterUserName
+      updatedAt
+      userAddress
```
Comment (clemsos) [inline unlock-app/src/hooks/useSocials.ts:86]: are all these fields needed ?

File: unlock-app/src/pages/_app.tsx [hunk 4]
```diff
@@ -38,7 +39,9 @@ const UnlockApp = ({ Component, pageProps }: AppProps) => {
           <ConnectModalProvider>
             <GlobalWrapper>
               <ErrorBoundary fallback={(props) => <ErrorFallback {...props} />}>
-                <Component pageProps={pageProps} />
+                <AirstackProvider apiKey={'1ef6142a6b64e48dd9fd4df8e0f4da9e3'}>
```
Comment (clemsos) [inline unlock-app/src/pages/_app.tsx:42]: :/ in clear

**Review by clemsos — APPROVED** [review]
few comments on my side

File: unlock-app/src/pages/_app.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/pages/_app.tsx:42]: unfortunately it is public... no way to not expose it.

File: unlock-app/src/hooks/useSocials.ts (hunk 3 above)
Comment (julien51) [reply unlock-app/src/hooks/useSocials.ts:86]: ha good point. not really. I will clean them up!

File: unlock-app/src/hooks/useEventOrganizers.ts (hunk 1 above)
Comment (julien51) [reply unlock-app/src/hooks/useEventOrganizers.ts:43]: I don't think we support that :(

**Pull request merged**